packages/core/src/bus/Event.ts

```diff
@@ -0,0 +1,30 @@
+import { EventDispatcher } from './EventDispatcher';
+import { getTypeName } from '../utils';
+import { v4 as uuidv4 } from 'uuid';
+
+export abstract class Event<T> {
+  public readonly type: string;
+  public readonly payload!: T;
+  public readonly correlationId!: string;
+  public readonly createdAt!: Date;
```

Comment:
You don't need to use null assertion since all of them are initialized in constructor
```suggestion
  public readonly payload: T;
  public readonly correlationId: string;
  public readonly createdAt: Date;
```